mttl/models/containers/__init__.py

```diff
@@ -300,38 +315,70 @@ def add_expert_to_transformer(
 
     for m_name, module in get_modules_to_modify_trie(transformer):
         if re.fullmatch(expert_config.modify_modules, m_name):
-            for c_name, layer in dict(module.named_children()).items():
-                if re.fullmatch(expert_config.modify_layers, c_name):
-                    total_layers += 1
-                    layer_name = f"{m_name}.{c_name}"
-
-                    if not isinstance(layer, ExpertContainer):
-                        CONTAINER_CLASS = get_container_class(model_modifier)
-                        expert_container = CONTAINER_CLASS(
-                            expert_config,
-                            layer,
-                            lora_merge_after=(
-                                selector_config.lora_merge_after
-                                if selector_config
-                                else False
-                            ),
-                        )
-                        expert_container.__layer_name__ = layer_name
-                        setattr(
-                            module,
-                            c_name,
-                            expert_container,
-                        )
-                        added_containers.append(expert_container)
-                    else:
-                        expert_container = layer
-
-                    added_layers.append(expert_container.__layer_name__)
-                    expert_container.add_expert(
-                        expert,
-                        action=action,
-                        is_default=is_default,
+            if len(expert_config.modify_layers) == 0:
```

Comment:
why?

Reply:
if modify_layers = "", we do not look into children of the module, and directly modify module instead of layers.

This is useful when we want to directly replace the full MLP module

Reply:
(nitpicking) I would check if `== ""` instead of `len == 0`

Reply:
or maybe checking if `modify_layers == None` ?